sw/host/opentitanlib/src/transport/cw310/mod.rs

```diff
@@ -203,6 +203,12 @@ impl Transport for CW310 {
             usb.pin_set_state(CW310::PIN_SW_STRAP0, true)?;
             usb.pin_set_state(CW310::PIN_SW_STRAP1, true)?;
             usb.pin_set_state(CW310::PIN_SW_STRAP2, true)?;
```

Comment:
these setting of defaults for the various strap pins (and the ones below) are only run if the fpga is re-programmed with a new bitstream, if it is determined that a new bitstream is not required (on line 196 above) we could return early without ever setting sensible defaults. Perhaps its worth moving all of these default pin state assignments to a separated function like the `init_direction()` function above (you could call it `init_pin_states` or something like that, and call this function  in the CW310 transport constructor, that way we start from a clean known state everytime.